packages/firebase_dynamic_links/firebase_dynamic_links/android/src/main/java/io/flutter/plugins/firebase/dynamiclinks/FlutterFirebaseDynamicLinksPlugin.java

```diff
@@ -239,7 +239,10 @@ private Task<Map<String, Object>> getDynamicLink(
               pendingDynamicLink =
                   Tasks.await(dynamicLinks.getDynamicLink(activity.get().getIntent()));
 
-              activity.get().getIntent().putExtra("flutterfire-used-link", true);
+              // Only save that the link was used if the activity is not detached already.
+              if (activity.get() != null || activity.get().getIntent() != null) {
```

Comment:
Shouldn't it be 
```suggestion
              if (activity.get() != null && activity.get().getIntent() != null) {
```

Reply:
Updated to the latter suggestion.

Looking again, should "flutterfire-used-link" be updated outside the (url != null) is/else starting on line 229?

Reply:
No, the url is used when we call `getDynamicLink(url)`. The intent is used when we call `getInitialLink()` which is a one time transaction. After it is used once, it will always return `null` . If the intent still exists, we check `flutterfire-used-link` to ensure it is not returned again. Here is the [PR and linked issue for context](https://github.com/firebase/flutterfire/pull/7743).